plugins/check_curl.c

```diff
@@ -324,9 +325,18 @@ int verify_callback(int preverify_ok, X509_STORE_CTX *x509_ctx)
 
 CURLcode sslctxfun(CURL *curl, SSL_CTX *sslctx, void *parm)
 {
-	(void) curl; // ignore unused parameter
-	(void) parm; // ignore unused parameter
-  SSL_CTX_set_verify(sslctx, SSL_VERIFY_PEER, verify_callback);
+  (void) curl; // ignore unused parameter
+  (void) parm; // ignore unused parameter
+  if(add_sslctx_verify_fun) {
```

Comment:
This is essentially a guard for when TLS backends do not provide verification or no TLS backend is available?

Reply:
cert verification depends on the ssl backend and curl version, so this function is not always required. Previously the callback had been added on demand for certificate verification, but there can only be a single callback. That's why i moved that outside
and use a switch to decide whether to add the verify function or not.